compiler/testData/codegen/boxInline/smap/rangeFoldingInClass.kt

```diff
@@ -0,0 +1,100 @@
+// FILE: 1.kt
+package test
+
+object A { inline fun f() {} }
+object B { inline fun g() {} }
+object C { inline fun h() {} }
+
+object D {
+    inline fun together() {
+        A.f()
+        C.h()
+        B.g()
+    }
+}
+
+// FILE: 2.kt
+import test.*
+
+object X {
+    // Unlike `rangeFolding.kt`, the calls in `D.together` refer to different
```

Comment:
@pyos Yes, it's expected.
There additional difference with spec: spec expects source path but compiler write class fqn. 
Not sure but maybe this behaviour should be revised

Reply:
Yeah, I was a bit confused by this part. Does IDEA use the class name for anything?

By the way, note that `KotlinDebug` could still be a single range, `generateDebugStrata` just has to sort ranges by `dest` and merge them separately from `generateDefaultStrata`.

Reply:
> Does IDEA use the class name for anything

No, it's Kotlin related artifact. Actually  there was a problem with determining source paths (not sure maybe smth was changed and it's possible to make some workaround to this lack now. cc @udalov)

> By the way, note that KotlinDebug could still be a single range, generateDebugStrata just has to sort ranges by dest and merge them separately from generateDefaultStrata.

Looks great. But best to postpone for implement it in another pr

It looks like a lack of debug/stepping test. Am right that it would be covered by #3214 ? It's best to have one or two related test in this pr too (if they are not present yet in master)

Reply:
What kind of stepping test do you mean? I don't think there's currently any kind of test at all other than the SMAP dumping ones that can possibly check KotlinDebug.

Reply:
Doesn't matter, I've missed that kt35006.kt is added in PR

Reply:
@pyos Could you add navigation test (KotlinExceptionFilterTestGenerated) for KT-35006
@NataliaUkhorskaya Could help with this type of tests

Reply:
Doesn't look like there are currently IDEA tests for this either (navigation tests are for clicking around in the source code), but I've added a NAVIGATE_TO_CALL_SITE directive to the exception filter tests (inlineFunCallSiteInInlineLambda.kt is KT-35006).